app/src/hooks/connectedWeb3.tsx

```diff
@@ -56,6 +56,8 @@ export const ConnectedWeb3: React.FC = props => {
       }
     } else if (error) {
       logger.debug(error.message)
+      localStorage.removeItem('CONNECTOR')
+      context.setConnector('Infura')
```

Comment:
When you cancel the metamask approval an error is thrown, and the `CONNECTOR` key isn't removed from the local storage, so when you refresh the modal for approval is shown again even before the `connect` button is clicked.